### DISCUSSION TITLE
Obtaining Github Actions performance metrics

### DISCUSSION THREAD

**Discussion by kmfrett** [community/community#160677] [Actions]
We need a way to capture metrics around our Github Actions jobs (run times, queue times, number of queued jobs, success/failure count, etc.) on a per-repo basis. We have found two potential options for getting these metrics: 

- Metrics exposed via the [action-metrics-server](https://github.com/actions/actions-runner-controller/blame/master/pkg/actionsmetrics/metrics.go#L14-L20) - introduced a year ago, and has no documentation. 
- An open-source [github-actions-exporter](https://github.com/Labbs/github-actions-exporter) for prometheus - option relies upon the Github API, which can cause us to eat into our API quota quite quickly by polling for these metrics every N seconds.

Is the action-metrics-server fully supported & production-ready? Are there any plans to add documentation around this?

Are there any recommendations for using the github-actions-exporter in such a way that it won’t burn our API quota too quickly?

Are there other options available for obtaining Github Actions metrics? (Do Github scale sets provide access to any metrics like this?)

**Comment by andrewmoyse-toast** [comment]
These are great questions. It's hard to rely on GitHub Actions for all the things if we can't observe whether they're working per repo.

**Comment by chlunde** [comment]
I would like `gha_available_jobs`, there's an issue here with no response: https://github.com/actions/actions-runner-controller/issues/3154

Also, the age of the oldest available job is the metric I would like to understand the user impact and detect if jobs are not started.

The other really important metric to understand the latency would be `gha_job_queue_duration_seconds` https://github.com/actions/actions-runner-controller/issues/3423
